controllers/cluster_controller.go

```diff
@@ -220,6 +220,13 @@ func (r *FoundationDBClusterReconciler) SetupWithManager(mgr ctrl.Manager, maxCo
 		return err
 	}
 
+	err = mgr.GetFieldIndexer().IndexField(context.Background(), &corev1.Node{}, "metadata.name", func(o client.Object) []string {
+		return []string{o.(*corev1.Node).Name}
```

Comment:
Why are we making this change? Overall, I think I am not following the controller related changes (a summary of why these changes are being made would be nice).

Reply:
Those where required by the the controller-runtime as they added some checks if all the requested fields have and indexer and when the node taint support was added, this was missed and the new controller-runtime cached this issue.